apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -245,6 +246,46 @@ const Ticket = new GQLListSchema('Ticket', {
         // `beforeChange` cannot be used, because data can be manipulated during updating process somewhere inside a ticket
         // We need a final result after update
         afterChange: async (...args) => {
+            const [operationArguments] = args
+            const { operation, updatedItem, context } = operationArguments
+
+            if (operation === 'create') {
+                const { property: propertyId } = updatedItem
+                const { data, errors: getPropertyErrors } = await context.executeGraphQL({
+                    query: Property.GET_ALL_OBJS_QUERY,
+                    variables: {
+                        where: {
+                            id: propertyId,
+                        },
+                    },
+                })
+
+                if (Array.isArray(getPropertyErrors) && getPropertyErrors.length) {
+                    throw new Error('[error]: Unable to increment rankIndicator while ticket is created.')
+                }
+
+                if (!Array.isArray(data.objs) && !data.objs.length) {
+                    throw new Error('[error]: Error while trying to get property query item.')
+                }
+
+                const [property] = data.objs
+                const { rankIndicator, dv, sender } = property
+
+                const { errors: updatePropertyErrors } = await context.executeGraphQL({
+                    query: Property.UPDATE_OBJ_MUTATION,
+                    variables: {
+                        id: property.id,
+                        data: {
+                            ...{ sender, dv },
+                            rankIndicator: rankIndicator + 1,
+                        },
+                    },
+                })
+
+                if (Array.isArray(updatePropertyErrors) && updatePropertyErrors.length) {
+                    throw new Error('[error]: Unable to increment rankIndicator while ticket is created.')
+                }
+            }
```

Comment:
It seems to me, when this logic will be extracted into separate module, it will get the code easier to maintain. Yes, it will be one-off usage of the module, but, the code will not be bloated.
If I, for example, would not use `storeChangesIfUpdated`, — it seems to me, that things will get very complex very quickly.
It's up to You, to extract or not, just a point to pay attention to.

Reply:
PS: such performance bottlenecks will soon or later get critical and will force us to love raw database engineering.
Sorry for off-top ;)

Reply:
It's good to extract it to some function to clarify the logic